provider/vsphere/provider_test.go

```diff
@@ -107,3 +112,83 @@ regions:
 	err = p.CloudSchema().Validate(v)
 	c.Assert(err, jc.ErrorIsNil)
 }
+
+type pingSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(pingSuite{})
+
+func (pingSuite) TestPingInvalidHost(c *gc.C) {
+	tests := []string{
+		"foo.com",
+		"http://IHopeNoOneEverBuysThisVerySpecificJujuDomainName.com",
```

Comment:
.invalid is expressly an invalid tld and I think example.com is also similarly designed for use. But foo.invalid is more what I would use here.